### ISSUE TITLE
CentOS 7 RPM repo and Python 2.7

### ISSUE THREAD

**Issue by jefmes** [overviewer/Minecraft-Overviewer#1204]
I'm running a pretty standard VM on DigitalOcean with CentOS 7, and I managed to get Overviewer working for a few months after building it myself. I was hoping by now there would be a CentOS 7 repo available and I'd really rather keep Overviewer updated via yum than have to use my kludgy mess. CentOS 7 ships with Python 2.7 also, so I'm getting the following message when I try to add the current repo from the CentOS instructions here:

http://docs.overviewer.org/en/latest/installing/

Here is the output I'm getting when attempting to install Overviewer via yum right now. Maybe I'm just missing something, but I thought someone might want to clean up the behavior for CentOS/RHEL 7

Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.pac-12.org
 * epel: mirror.sfo12.us.leaseweb.net
 * extras: mirror.lax.hugeserver.com
 * updates: mirrors.psychz.net
Resolving Dependencies
--> Running transaction check
---> Package Minecraft-Overviewer.x86_64 0:0.12.31-1.el6 will be installed
--> Processing Dependency: python(abi) = 2.6 for package: Minecraft-Overviewer-0.12.31-1.el6.x86_64
--> Processing Dependency: libpython2.6.so.1.0()(64bit) for package: Minecraft-Overviewer-0.12.31-1.el6.x86_64
--> Finished Dependency Resolution
Error: Package: Minecraft-Overviewer-0.12.31-1.el6.x86_64 (overviewer)
           Requires: python(abi) = 2.6
           Installed: python-2.7.5-16.el7.x86_64 (@anaconda)
               python(abi) = 2.7
               python(abi) = 2.7
Error: Package: Minecraft-Overviewer-0.12.31-1.el6.x86_64 (overviewer)
           Requires: libpython2.6.so.1.0()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by jefmes** [comment]
Just to follow up, when using --skip-broken as it suggests, it fails differently.

Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.pac-12.org
 * epel: mirror.sfo12.us.leaseweb.net
 * extras: mirror.lax.hugeserver.com
 * updates: mirrors.psychz.net
Resolving Dependencies
--> Running transaction check
---> Package Minecraft-Overviewer.x86_64 0:0.12.31-1.el6 will be installed
--> Processing Dependency: python(abi) = 2.6 for package: Minecraft-Overviewer-0                          .12.31-1.el6.x86_64
--> Processing Dependency: libpython2.6.so.1.0()(64bit) for package: Minecraft-O                          verviewer-0.12.31-1.el6.x86_64
http://overviewer.org/rpms/7/x86_64/repodata/filelists.xml.gz: [Errno 14] HTTP E                          rror 404 - Not Found
Trying other mirror.


 One of the configured repositories failed (Minecraft-Overviewer - 7 - x86_64),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Disable the repository, so yum won't use it by default. Yum will then
        just ignore the repository until you permanently enable it again or use
        --enablerepo for temporary usage:

            yum-config-manager --disable overviewer

     4. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=overviewer.skip_if_unavailable=tr                          ue

failure: repodata/filelists.xml.gz from overviewer: [Errno 256] No more mirrors                           to try.
http://overviewer.org/rpms/7/x86_64/repodata/filelists.xml.gz: [Errno 14] HTTP E                          rror 404 - Not Found

**Comment by jefmes** [comment]
Sorry for the ignorance, I'm still getting familiar with GitHub. :)  So how does this work? I imagine not anyone can jump in and fix this, because there is likely a limited number of people who have access to update the RPM repodata. Is there anything I can do to help, or is there more data that needs to be provided? I noticed there isn't even a directory for 7. If someone can take care of that, I'd be happy to try it again and see if I get different results.

![overviewerrpms](https://cloud.githubusercontent.com/assets/5678101/6142253/54f2c0c6-b167-11e4-9675-86d8ae0e18b3.PNG)

**Comment by eminence** [comment]
I think what we need to do is setup a new rpm builder for CentOS7.  @aheadley should hopefully look at this soon.  You're right, this isn't really an Overviewer code issue, something needs to change in our backend build infrastructure.  If someone really wants to get their hands dirty, I think what we need is an investigation about why this python26 dependency is coming from.   Ultimately I think we need a new spec file for CentOS/RHEL 7.  @aheadley can yell at me if I'm off track, though.

In the meantime, you should be able to easily build Overviewer from source.   Check the docs for details and let us know if you run into any issues

**Comment by jefmes** [comment]
Thanks, greatly appreciate the response! I'll go ahead and recompile it for my server for now then and check in later.

**Comment by louiscbrooks** [comment]
I was able to install on CentOS 7 using the Fedora 19 repo.

**Comment by Luukth** [comment]
Can someone maybe tell me how I can install overviewer on my VPS with centOS 7 activated??

**Comment by Luukth** [comment]
I tried this out: http://docs.overviewer.org/en/latest/building/#centos-5

But when I try the command python setup.py build.

I get this error:
[root@play Minecraft-Overviewer]# python setup.py build                         running build
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/util.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/nbt.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/logger.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/rendermodes.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/progressbar.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/items.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/optimizeimages.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/cache.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/__init__.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/observer.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/overviewer_version.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/settingsValidators.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/tileset.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/textures.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/configParser.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/settingsDefinition.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/files.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/assetmanager.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/rcon.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/world.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/signals.py -> build/lib.linux-x86_64-2.7/overviewer_core
copying overviewer_core/dispatcher.py -> build/lib.linux-x86_64-2.7/overviewer_core
creating build/lib.linux-x86_64-2.7/overviewer_core/aux_files
copying overviewer_core/aux_files/genPOI.py -> build/lib.linux-x86_64-2.7/overviewer_core/aux_files
copying overviewer_core/aux_files/__init__.py -> build/lib.linux-x86_64-2.7/overviewer_core/aux_files
creating build/lib.linux-x86_64-2.7/overviewer_core/data
creating build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/grass.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/portal.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/lava.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/foliage.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/ender.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/normal_double.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/fire.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/light_normal.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/watercolor.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/chest.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
copying overviewer_core/data/textures/water.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/textures
creating build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
creating build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_tower_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_anvil.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_hoe.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_base_plain.svg -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_ship.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_town.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_hoe_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_town_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_anvil_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_factory.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_factory_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_ship_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_base_plain_red.svg -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_tower.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_mine.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/icons/marker_mine_red.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets/icons
copying overviewer_core/data/web_assets/compass_upper-right.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/signpost-shadow.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/backbone.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/overviewer.css -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/control-bg-active.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/regions.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/index.html -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/control-bg.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/bed.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/compass_lower-right.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/underscore.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/compass_lower-left.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/compass_upper-left.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/signpost_icon.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
copying overviewer_core/data/web_assets/signpost.png -> build/lib.linux-x86_64-2.7/overviewer_core/data/web_assets
creating build/lib.linux-x86_64-2.7/overviewer_core/data/js_src
copying overviewer_core/data/js_src/overviewer.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/js_src
copying overviewer_core/data/js_src/util.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/js_src
copying overviewer_core/data/js_src/views.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/js_src
copying overviewer_core/data/js_src/models.js -> build/lib.linux-x86_64-2.7/overviewer_core/data/js_src
running build_ext
building 'overviewer_core.c_overviewer' extension
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/overviewer_core
creating build/temp.linux-x86_64-2.7/overviewer_core/src
creating build/temp.linux-x86_64-2.7/overviewer_core/src/primitives
gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I. -I/usr/lib64/python2.7/site-packages/numpy/core/include -I/usr/include/python2.7 -c overviewer_core/src/main.c -o build/temp.linux-x86_64-2.7/overviewer_core/src/main.o -Wno-unused-variable -Wno-unused-function -Wdeclaration-after-statement -Werror=declaration-after-statement
In file included from /usr/lib64/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1728:0,
                 from /usr/lib64/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:17,
                 from /usr/lib64/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:15,
                 from overviewer_core/src/overviewer.h:33,
                 from overviewer_core/src/main.c:18:
/usr/lib64/python2.7/site-packages/numpy/core/include/numpy/npy_deprecated_api.h:11:2: warning: #warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
  ^
In file included from overviewer_core/src/main.c:18:0:
overviewer_core/src/overviewer.h:34:21: fatal error: Imaging.h: No such file or directory
 #include <Imaging.h>
                     ^
compilation terminated.
Traceback (most recent call last):
  File "setup.py", line 273, in run
    build.run(self)
CompileError: command 'gcc' failed with exit status 1

Failed to build Overviewer!
Please review the errors printed above and the build instructions
at <http://docs.overviewer.org/en/latest/building/>.  If you are
still having build problems, file an incident on the github tracker
or find us in IRC.

**Comment by eminence** [comment]
This error (Imaging.h: No such file or directory) means you're missing the `python-imaging-devel` package

**Comment by Luukth** [comment]
And how can I use the correct one??

**Comment by CounterPillow** [comment]
Install the package `python-pillow-devel` with yum.

**Comment by Luukth** [comment]
Alright gone try that right now. thnx for the tip

**Comment by Luukth** [comment]
wow thanks it worked. Thank you so much!

**Comment by mianosm** [comment]
Fedora 22 / EL7 rpm directories would be highly appreciated! :+1:

**Comment by karpana** [comment]
EL7 rpm directories would be greatly appreciated.  
I managed to get overviewer installed in January 2015, but am unable to update.

I've even tried to consume the EL6 rpm directories with the same resulting installation failures.

**Comment by MarkehMe** [comment]
Your best bet is to install it from source (ugh)

Like so...

```bash
yum install -y python-imaging python-pillow-devel python-devel numpy

cd /usr/local/src

git clone git://github.com/overviewer/Minecraft-Overviewer.git 

# Give this a bit to grab the repo, you will find it easier to update if you grab the repo and not just a zip from github

cd Minecraft-Overviewer

python setup.py build
```

**Comment by enaut** [comment]
I did setup a copr repository for Fedora!
https://copr.fedorainfracloud.org/coprs/enaut/minecraft-overviewer-weekly/
As I cannot test CentOS I did not enable those so far but as soon as someone is up to do the Testing and debugging I'm happy to enable epel builds (I did it).

https://copr.fedorainfracloud.org/coprs/enaut/minecraft-overviewer-weekly/

I'm doing the updates manually whenever something is committed to overviewer. The tito branch with the specfile source is here: https://github.com/enaut/Minecraft-Overviewer/tree/weekly-tito

**Comment by rx7raven** [comment]
I used the rep @enaut setup and I'll let you know how it works.  Running CentOS 7.

**Comment by rrimc69** [comment]
Still no sane path for el7 users... I just wont go Ubuntu for this, so I suppose I'm stuck on el6, and will have to restore my backup image.

Given the date I assume it's past an effort. Wanted to keep my minecraft server OS up to date, but hardly think el7 is bleeding edge ... 

Manual install does not work as expected either, using the exact same process described above, however doubly frustrating as this is a  dependency of sandain's mscs script hosted elsewhere here... which equally does not play nice on el7.

**Comment by CounterPillow** [comment]
@rrimc69 just build from source, no need to get all melodramatic.

**Comment by enaut** [comment]
@rrimc69 Did you try the repos from copr I created?
https://copr.fedorainfracloud.org/coprs/enaut/minecraft-overviewer-weekly/

**Comment by rrimc69** [comment]
@enaut, no I haven't, but thanks for the suggestion. I'm only now starting migration of all EL5/6 kit at work to El7, so worth the hunt & fix, and kind of fun looking at newer documentation with systemctl too. 

I did find the root cause of my issues with mscs and overviewer, during an further re-build. The cause is EL7 sudo's path re-interpretation behavior and overviewers' manual scripted install location, along with the way mscs calls it. 

For further details see here: https://github.com/MinecraftServerControl/mscs/issues/142 

@CounterPillow  I accept I may not have been very clear with my text, 'Manual install'  ~ 'install from source'. I did build my initial VM from source and overviewer clearly has EL7 issues, as does the mscs script.

You may wish to review the link above, and reflect on the content. It may assist with bringing overviewer.py in line with the practices adhered to by the vast majority of the code I've deployed at work, or home, be it Windows, Linux or BSD, that just work 'out of the box' for the intended use, by the right user. 

It is interesting you find 'Articulate' is synonymous with 'Melodramatic' ... User frustration is clearly lost in translation for you. I'm not a kid learning coding, so by all means ... play the man not the ball, and continue to complain about my felicity of style. You'll only further expose your intellect by doing so.

Cheers,

Rory

**Comment by MarkehMe** [comment]
@CounterPillow I wouldn't say that was melodramatic.. it's frustrating. As I'm sure you've realised.  

CentOS 6 is actually stopping its full updates next year and moving to maintenance updates only. CentOS 5 stopped receiving full updates in 2014 and is currently on maintenance updates.. and those end early next year. 😨 

I think we should come back to the original issue of this ticket and simply add a package for CentOS 7. If you would like, I'm happy to help you do that. 😃

**Comment by agrif** [comment]
What are the chances that symlinking a CentOS 7 directory would work? Can somebody on CentOS 7 try the most recent CentOS 6 RPM?

**Comment by CounterPillow** [comment]
@rrimc69 
>I did build my initial VM from source and overviewer clearly has EL7 issues

Mind sharing what those issues are then, if they're unrelated to third-party scripts and missing binary packages from our side? Because as far as I can see, overviewer should run just fine from source. If you're having trouble with a third party script making improper assumptions about installation paths, then that's not an overviewer issue.

**Comment by rrimc69** [comment]
Ok.. 
-bash-4.2$ whoami
minecraft
-bash-4.2$ which /usr/local/bin/mscs
/usr/local/bin/mscs
-bash-4.2$ which /usr/local/bin/msctl
/usr/local/bin/msctl
-bash-4.2$ mscs create wembledon 25567 10.0.0.99
which: no overviewer.py in (/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
Creating Minecraft world: wembledon.
-bash-4.2$ mscs status
which: no overviewer.py in (/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
Minecraft Server Status:
  wembledon: not running.
  wombledon: not running.
-bash-4.2$ exit

**Comment by CounterPillow** [comment]
@rrimc69 could you show me the steps where you actually build and install overviewer from source? I don't know what mscs is, and just because mscs can't find overviewer doesn't mean the problem lies with overviewer.

**Comment by rrimc69** [comment]
Sudo behaviour as described here in un-related works: http://serverfault.com/questions/528096/sudo-passenger-memory-stats-command-not-found

My Host:
[root@mine2 worlds]#  sudo sh -c 'echo $PATH'
/sbin:/bin:/usr/sbin:/usr/bin
[root@mine2 worlds]# which sudo
/usr/bin/sudo
[root@mine2 worlds]# cat /etc/redhat-release
CentOS Linux release 7.2.1511 (Core)
[root@mine2 worlds]#

**Comment by rrimc69** [comment]
Here..
[OverviewerBuildOutput.txt](https://github.com/overviewer/Minecraft-Overviewer/files/605471/OverviewerBuildOutput.txt)

**Comment by rrimc69** [comment]
You've clearly not followed the link... or read the text.

**Comment by CounterPillow** [comment]
>OverviewerBuildOutput.txt
All you do is build, not install. How should overviewer.py be present in $PATH when you don't do a local install?

Also, I'd appreciate it if you were to adopt a less hostile tone.

**Comment by agrif** [comment]
@rrimc69 you have built Overviewer, but not installed it, technically. You would need to run `setup.py install` to do that, but I **don't recommend it** (it could conflict with the CentOS 7 RPMs when we fix those).

Overviewer will happily run from `/opt/Minecraft-Overviewer`. You just have to add that to your `$PATH`.

I don't know why, but `mscs` is running Overviewer as root. So you'll need to add `/opt/Minecraft-Overviewer` to root's `$PATH` as well. I would stick it at the end of your `$PATH`, so that when the RPMs are fixed, the RPM version will take precedence.

**Comment by rrimc69** [comment]
I suppose I could sudo run the installer but that is not the advice up the stream.  

Review your documentation ... right now..
http://docs.overviewer.org/en/latest/building/#linux

No example of .. 'install' given.  Nor in the the mention of source build above. Shan't be accused of not reading the manual now...

**Comment by agrif** [comment]
@rrimc69 we do not mention `install` because we don't recommend doing it. Like I said, it can break actual prebuilt packages.

When you build from source, you're expected to run it from the place you extracted it. As mentioned in the next section, [Running the Overviewer](http://docs.overviewer.org/en/latest/running/#usage)...

> For Mac or Linux builds from source, you would run something like this with the current directory in the top level of the source tree:
> `./overviewer.py /opt/minecraft/server/world /opt/minecraft/mcmap`

The confusion seems to be coming from the fact that `mscs` doesn't do that, or provide a way to do that. That's not something in *our* control (you'll have to take it up with the mscs guys) but it is something you can fix by putting the Overviewer source directory in your `$PATH`.

**Comment by rrimc69** [comment]
@agrif
Granted it works if directly called, you will see I've already stated that it commenced for me as a sudo path issue, and offered that feedback to the mscs author. Still an issue for EL7 overview users however who may well expect the installation binary to be in one of the standard PATH locations.

Your OP user may well have bumped that issue and not commented. I have, at least if they read this thread they'll 'get it' if they're regular EL6/7 users.

It may well be I've not 'installed' it, but it appears the documentation for a Linux manual install does not show this step, and it is clearly not inherent knowledge that it requires an install invocation. A job for the installer, yes I do see your point. However, in fairness, I'm on Linux not OSX so not likely to read OSX install text.. nor would most time poor sysadmins with bad habits, and .. language issues .. :)

**Comment by agrif** [comment]
It was not our intention that the "Build from Source" section of our documentation be seen as an installation guide. We deliberately chose to use "install" to refer to the prebuilt binaries, and only say "build" for the source-based method, since no install actually ever happens.

I'm sorry if this caused confusion.

**Comment by CounterPillow** [comment]
The local install step is the same as with any python distutils/setuptools package. And again, the problem comes down to you not setting your $PATH variable properly, as we discourage from doing local installs due to the problems associated with them down the road.

Yes, binary packages would be nice to have, but insulting contributors and acting like a jerk here won't make anyone more motivated to sacrifice some of their free time to set up a system to package something for an operating system they don't even use.

**Comment by rrimc69** [comment]
@agrif thanks for your measured input. I reviewed what you said, yes I have not installed it. It may be In the absence of a packaged solution, the install from source documentation is the eventual outcome, so may be more focus is needed there.

I generate the openproject work packages for all ICT in my org, and that tool has a workflow functionality. (Beats server built/admin tasks at the moment..) One particular tool used in that documentation review is 'use case' focused reviews. I normally schedule documentation review after code review in my project plans, It works and normally, this finds holes and weaknesses in the existing documentation. An automated process that, or something similar, if git hub supports it, may be useful to you. 

You may still wish to reflect on or consider the sudo behavior as part of your EL7 packaging effort, the /usr/local/bin location will not be found by sudo, either, if used for your package. 

To test, I removed my symlink and ran `python setup.py install' it completed OK as far as I can see, can't provide the text output, I accidently dropped the shell and lost the buffer, sorry. Binary is available afterwards to user, and 3rd party script alike:

[rory@mine2 ~]$ which overviewer.py
/usr/local/bin/overviewer.py
[rory@mine2 ~]$ sudo mscs status
[sudo] password for rory:
Minecraft Server Status:
  2ring: running.
    Process ID: 24400.
    Memory used: 529924 kB.
  landoffun: running.
    Process ID: 24570.
    Memory used: 428836 kB.
  oblivion: running.
    Process ID: 24749.
    Memory used: 534480 kB.
  r3dbird: running.
    Process ID: 24928.
    Memory used: 442248 kB.
  wembledon: running version 1.11 (0 of 20 users online).
    Process ID: 25216.
    Memory used: 577140 kB.
  wombledon: running version 1.11 (0 of 20 users online).
    Process ID: 25514.
    Memory used: 974928 kB.
[rory@mine2 ~]$

Cheers, 

Rory.

**Comment by rrimc69** [comment]
Oh .. and thanks belatedly.

**Comment by agrif** [comment]
`setup.py install` will default to `/usr/local/bin`, in rather the same way any GNU autotools packages do. However, the rpm packages (IIRC) will install to `/usr/bin`, as you would expect from a proper package.

Not having a package for CentOS 7 sucks, we know. Most of us have been terrifically busy (for instance, I started grad school) and don't have time to work on Overviewer like we used to. That said, [a fix is coming down the pipes](https://github.com/overviewer/Overviewer-Web/compare/centos7), so bear with us.

**Comment by rrimc69** [comment]
lol.. thats fine agrif, I get busy and complex, I do it for a living, I can relate. 

I've no personal expectation of instant fix though, my box works now, and I can build host and populate worlds from backup, using scripts/pxe menu in about 20 minutes. Think I'll leave automated renders till there is an rpm though. 

I hope this discussion helps in some small way to get your EL7 solution closer to done.

**Comment by rrimc69** [comment]
Note for @jefmes, if your still tuned in, hope this helps, my eventual working build was brutally minimal no desktop, epel was required for other reasons, below are the just python related rpms as at Nov '17, I note there is no 2.6 in there, with full rpm attached if you need to dive deeper.
[Mine2-rpm-list.txt](https://github.com/overviewer/Minecraft-Overviewer/files/605968/Mine2-rpm-list.txt)

Just the python kit:
[rory@mine2 ~]$ yum list installed | grep python
dbus-python.x86_64                  1.1.1-9.el7                        @anaconda
libselinux-python.x86_64            2.2.2-6.el7                        @anaconda
newt-python.x86_64                  0.52.15-4.el7                      @anaconda
python.x86_64                       2.7.5-39.el7_2                     @updates
python-backports.x86_64             1.0-8.el7                          @base
python-backports-ssl_match_hostname.noarch
python-configobj.noarch             4.7.2-7.el7                        @anaconda
python-decorator.noarch             3.4.0-3.el7                        @anaconda
python-devel.x86_64                 2.7.5-39.el7_2                     @updates
python-iniparse.noarch              0.4-9.el7                          @anaconda
python-javapackages.noarch          3.4.1-11.el7                       @base
python-libs.x86_64                  2.7.5-39.el7_2                     @updates
python-lxml.x86_64                  3.2.1-4.el7                        @base
python-nose.noarch                  1.3.0-3.el7                        @base
python-perf.x86_64                  3.10.0-327.36.3.el7                @updates
python-pillow.x86_64                2.0.0-19.gitd1c6db8.el7            @base
python-pillow-devel.x86_64          2.0.0-19.gitd1c6db8.el7            @base
python-pip.noarch                   7.1.0-1.el7                        @epel
python-pycurl.x86_64                7.19.0-17.el7                      @anaconda
python-pyudev.noarch                0.15-7.el7_2.1                     @updates
python-setuptools.noarch            0.9.8-4.el7                        @base
python-slip.noarch                  0.4.0-2.el7                        @anaconda
python-slip-dbus.noarch             0.4.0-2.el7                        @anaconda
python-urlgrabber.noarch            3.10-7.el7                         @anaconda
rpm-python.x86_64                   4.11.3-17.el7                      @anaconda
[rory@mine2 ~]$

Cheers,

Rory.

**Comment by eminence** [comment]
We've been making slow but steady progress on this, and we are now building RPMs for centos7.  A RPM repository is not yet set up yet, but you can find the latest RPM for centos7 here:

http://static-backend.overviewer.org/builds/master/681/Minecraft-Overviewer-0.12.175-1.el7.x86_64.rpm

This corresponds to the latest master (at the time of this writing, 63df7976814e297175ad0b17ebb4112557d9235b)

**Comment by MarkehMe** [comment]
Great to hear!

**Comment by rrimc69** [comment]
Awsome. sounds great, now all I need is time to go test :)  employer has
had me full capacity for 2 months now .. holidays soon !!

**Comment by seifer44** [comment]
Soooooooooo are you guys supporting EL7 yet?

**Comment by CounterPillow** [comment]
@seifer44 there's no problem with the code, you can run Overviewer on EL7 right now, and have been able to for years.

There's just no repo of prebuilt binary packages.

**Comment by seifer44** [comment]
@Markehme nailed it. I see EL6 is supported, and on another thread somewhere around here, EL7 is supported by subscribing to latest for the version instead of 7. Also, [this page](http://overviewer.org/downloads) makes it seem like EL7 isn't supported at all.

On top of that, the packages in the latest repository don't even include the 1.12 version support. I had to go find the package manually [here](http://static-backend.overviewer.org/builds/master/).

With all of that in mind, it doesn't seem very supported to me.

**Comment by CounterPillow** [comment]
@MarkehMe I'm not being stubborn, I just don't use any RPM-based distros and have nothing to do with the build infrastructure and direct people to the appropriate solution for the time being, which is building from source, which is really not all that hard.

Please stop acting like this is somehow a big company's product you're paying for and can make demands to, because this is really just a bunch of dudes doing stuff in their free time, and you're basically telling them to use their free time in a way that benefits you without spending any of your own free time on this.

**Comment by MarkehMe** [comment]
@CounterPillow sure we won't call it _stubborn_ 😉 - however if you reread your past comments you sure don't sound like you even want to even push for it. 

**No one is acting like it's a big company product we're paying for**. It doesn't benefit me anymore, we're all just getting impatient.

When an easy fix issue like this goes on for over 2 and a half years people are going to be annoyed - obviously - and complain, and you're the only one responding so yes, you're going to cop a mouthful from people. Your efforts are appreciated, but our responses are nothing but human. 

 ```
          ¯\_(ツ)_/¯
```

**Comment by CounterPillow** [comment]
Then here's things you can do:

* host build infrastructure and a repository yourself
* submit a PR to the Overviewer-Web repository
* go on IRC and talk with achin about what needs to be done to get the new build infra going

anything else (including shitty reaction GIFs in github issues) is a waste of time.

**Comment by seifer44** [comment]
I think you may be mistaken. CentOS is part of the Enterprise Linux (EL) definition. CentOS, being free, is used by a lot of individuals running servers (including me).

This got toxic. How about I offer to help, instead. Is there an automation tool that you guys use? It might not be that hard to push stuff to all of those places that I mentioned. Also, is there any serious difference to the package for the other RPM supported distributions? If not, maybe remove the distro part of the package name in the SPEC file.

**Comment by MarkehMe** [comment]
You don't have to get a temper over this. I didn't have a clue the _web repository_ existed or that a IRC channel existed. I can work on a PR for it when I worked out how it all works. Thank you for sharing it, it's appreciated. 

That could have been something you should have [contributed earlier in the discussion](#issuecomment-260194217) and we wouldn't have had all this time wasted going back and forth (so don't bad mouth me for using visual communications haha). 😃

**Comment by CounterPillow** [comment]
>I think you may be mistaken. CentOS is part of the Enterprise Linux (EL) definition. CentOS, being free, is used by a lot of individuals running servers (including me).

where did I ever claim this wasn't the case?

>That could have been something you should have contributed earlier in the discussion and we wouldn't have had all this time wasted going back and forth

Usually there's no need to tell people explicitly that you're able to contribute to open-source projects, but apparently you need to be spoonfed this.

**Comment by eminence** [comment]
Hi everyone, I know we have been massively slacking on this topic.    We've started work on this a while ago, but never really finished.  You can preview the new RPM repo here: http://overviewer.org/rpms_new/ (though this link will eventually be removed once it's officially hooked up to the official repo -- whenever that might be).



I know that people can get frustrated with the lack of progress, and that people can get frustrated at the lack of help.  Please keep our discussion threads friendly!  :pizza:

**Comment by enaut** [comment]
Hey Guys I once created a copr repos since there seemed the desperate need for one: https://copr.fedorainfracloud.org/coprs/enaut/minecraft-overviewer-weekly/
Inclusion in the official docs was not wanted (I think that's where the stubborn comes from).

Since I had next to none downloads I stopped maintaining it as I don't really use it myself. But hey it might be worth to look at and might be easy enough to just maintain it yourself starting a build is like 3 to 5 commands and then you have your rpm repos included. The relevant branch is: https://github.com/enaut/Minecraft-Overviewer/tree/weekly-tito it was my first copr package so the history is a little cluttered I think but hey it works and is in shape.

If anyone wants to step in I will gladly provide the exact commands etc for building.
steps are:
 1. bump the version (tito something)
 2. git commit 
 3. git add a tag
 4. and then from the copr interface restart the build

**Comment by seifer44** [comment]
@eminence anything I can do to help with that? Test case, setup, whatever.

**Comment by agrif** [comment]
Fundamentally the issue is that none of the core developers of Overviewer use it anymore, and so fixing anything more than a trivial issue sometimes has to wait for our free time and motivation to align.

I will *try* to get an EL7 repo working this weekend. If I fail, contributions are welcome.

**Comment by enaut** [comment]
# Adding a new repos to the copr repos with tito:

The relevant git branches:
```
$ git branch -a
* weekly-tito
  remotes/enaut/weekly-tito
  remotes/overviewer/master
  [ ... ]
```
Getting the tito branch:
`$ git checkout weekly-tito`

Fetching the new master:
`$ git fetch overviewer master`

Merging the new commits to the `weekly-tito` branch when said branch is checked out:
`$ git merge overviewer/master`

Making a new versionbump so dnf will know that this package is newer than before:
```
$ tito tag
# editor opens with ready commit message (including changelog generated from git log)
Creating output directory: /tmp/tito
Tagging new version of minecraft-overviewer: 0.12-11 -> 0.12-12
Created tag: minecraft-overviewer-0.12-12
   View: git show HEAD
   Undo: tito tag -u
   Push: git push origin && git push origin minecraft-overviewer-0.12-12
```
Do the suggested:
```
$ git push
$ git push minecraft-overviewer-0.12-12
```
Open https://copr.fedorainfracloud.org :
In the repository add a build with the settings:
![bildschirmfoto von 2017-08-16 08-09-05](https://user-images.githubusercontent.com/290005/29349462-583ff656-825a-11e7-8570-6239982820b1.png)

click submit! Now its building and aft 10 minutes you have your repository.

# Installing from copr repos

Note that the copr repos are well integrated into the os for example you ca search for something without having any of the repositories installed:
```
$ sudo dnf copr search minecraft-overviewer
======================== Gefunden: minecraft-overviewer ========================
enaut/minecraft-overviewer-weekly : Minecraft-Overviewer - A map renderer for
     ...: Minecraft
     ...:  * This repository is updated to the most recent git version every
     ...:    week
```
installing is as simple as:
```
    # dnf copr enable enaut/minecraft-overviewer-weekly
    # dnf install minecraft-overviewer
```
execute with:

    > overviewer.py <world-directory> <target-directory>

the ease of using copr repos tops the "I have my own repos on my own server by far - in my opinion" but last I tried I had the impression that kind of repos is not welcome here. - I proposed updating the docs etc. but it was said: "we'd rather have this on our own infrastructure"

**Comment by agrif** [comment]
I mean, we *would* like to have it in our system. We're not living up to that ideal though. If I can't get our system working with it this weekend, I'll talk to the others about adding yours to the docs.

(If I forget, feel free to poke me about it!)
